lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -401,65 +412,78 @@ GroupDoneOp GroupOp::getDoneOp() {
 }
 
 //===----------------------------------------------------------------------===//
-// CellOp
+// Utilities for Cell-like operations.
+//===----------------------------------------------------------------------===//
+
+/// Gives each result of the cell a meaningful name in the form:
+/// <prefix>.<port-name>
+static void getCellAsmResultNames(OpAsmSetValueNameFn setNameFn, Operation *op,
+                                  ArrayRef<StringRef> portNames) {
+  assert(op->hasTrait<Cell>() && "must have the Cell trait");
+
+  auto instanceName = op->getAttrOfType<StringAttr>("instanceName").getValue();
+  std::string prefix = instanceName.str() + ".";
+  for (size_t i = 0, e = portNames.size(); i != e; ++i)
+    setNameFn(op->getResult(i), prefix + portNames[i].str());
+}
+
+//===----------------------------------------------------------------------===//
+// InstanceOp
 //===----------------------------------------------------------------------===//
 
 /// Lookup the component for the symbol. This returns null on
 /// invalid IR.
-ComponentOp CellOp::getReferencedComponent() {
+ComponentOp InstanceOp::getReferencedComponent() {
   auto program = (*this)->getParentOfType<ProgramOp>();
   if (!program)
     return nullptr;
 
   return program.lookupSymbol<ComponentOp>(componentName());
```

Comment:
I was wondering whether this is safe? The `ComponentOp` parent of the `InstanceOp` has trait `IsolatedFromAbove` which makes me think that accessing the enclosing ProgramOp's symbol table is unsafe. I am still new to MLIR so maybe I just misunderstand the legality implications of `IsolatedFromAbove`!

Reply:
IsolatedFromAbove only refers to SSA values.  Indeed, one of the reasons behind the need for symbol tables is to provide an explicit mechanism for regions that are isolatedFromAbove (like a region inside a function definition) to refer to other code (like another function definition in a call).  see https://mlir.llvm.org/docs/Traits/#isolatedfromabove